astropy/io/fits/tests/test_diff.py

```diff
@@ -113,9 +113,37 @@ def test_floating_point_tolerance(self):
         assert not diff.identical
         assert (diff.diff_keyword_values ==
                 {'B': [(2.00001, 2.00002)], 'C': [(3.000001, 3.000002)]})
-        diff = HeaderDiff(ha, hb, tolerance=1e-6)
+        diff = HeaderDiff(ha, hb, rtol=1e-6)
         assert not diff.identical
         assert diff.diff_keyword_values == {'B': [(2.00001, 2.00002)]}
+        diff = HeaderDiff(ha, hb, rtol=1e-5)
+        assert diff.identical
+
+    def test_floating_point_atol(self):
+        ha = Header([('A', 1), ('B', 1.0), ('C', 0.0)])
+        hb = ha.copy()
+        hb['B'] = 1.00001
+        hb['C'] = 0.000001
+        diff = HeaderDiff(ha, hb, rtol=1e-6)
+        assert not diff.identical
+        assert (diff.diff_keyword_values ==
+                {'B': [(1.0, 1.00001)], 'C': [(0.0, 0.000001)]})
+        diff = HeaderDiff(ha, hb, rtol=1e-5)
+        assert not diff.identical
+        assert (diff.diff_keyword_values ==
+                {'C': [(0.0, 0.000001)]})
+        diff = HeaderDiff(ha, hb, atol=1e-6)
+        assert not diff.identical
+        assert (diff.diff_keyword_values ==
+                {'B': [(1.0, 1.00001)]})
+        diff = HeaderDiff(ha, hb, atol=1e-5)  # strict inequality
+        assert not diff.identical
+        assert (diff.diff_keyword_values ==
+                {'B': [(1.0, 1.00001)]})
+        diff = HeaderDiff(ha, hb, rtol=1e-5, atol=1e-5)
+        assert diff.identical
+        diff = HeaderDiff(ha, hb, atol=1.1e-5)
+        assert diff.identical
```

Comment:
Yes, I would like to see such a test as well, where both `rtol` and `tolerance` are defined, but the latter overrides the former.